docs/grammar/Solidity.g4

```diff
@@ -0,0 +1,489 @@
+ * An emit statement. The contained expression needs to refer to an event.
+ */
+emitStatement: Emit expression callArgumentList Semicolon;
+/**
+ * An inline assembly block.
+ * The contents of an inline assembly block use a separate scanner/lexer, i.e. the set of keywords and
+ * allowed identifiers is different inline an inline assembly block.
+ */
+assemblyStatement: Assembly AssemblyDialect? AssemblyLBrace yulStatement* YulRBrace;
+
+//@doc:inline
+variableDeclarationList: variableDeclarations+=variableDeclaration (Comma variableDeclarations+=variableDeclaration)*;
+/**
+ * A tuple of variable names to be used in variable declarations.
+ * May contain empty fields.
+ */
+variableDeclarationTuple:
+    LParen
+        (Comma* variableDeclarations+=variableDeclaration)
+        (Comma (variableDeclarations+=variableDeclaration)?)*
+    RParen;
+/**
+ * The declaration of one with an optional initial value or a tuple of variable with required initial value.
+ */
+variableDeclarationStatement: ((variableDeclaration (Assign expression)?) | (variableDeclarationTuple Assign expression)) Semicolon;
+expressionStatement: expression Semicolon;
+
+/**
+ * The catch clause of a try statement.
+ */
+catchClause: Catch (identifier? LParen (arguments=parameterList) RParen)? block;
+
+mappingType: Mapping LParen key=mappingKeyType Arrow value=typeName RParen;
+/**
+ * Only elementary types or user defined types are viable as mapping keys.
+ */
+mappingKeyType: elementaryTypeName[false] | userDefinedTypeName;
+
+/**
+ * A Yul statement within an inline assembly block.
+ * continue and break statements are only valid within for loops.
+ * leave statements are only valid within function bodies.
+ */
+yulStatement:
+    yulBlock
+	| yulVariableDeclaration
+	| yulAssignment
+	| yulFunctionCall
+	| yulIfStatement
+	| yulForStatement
+	| yulSwitchStatement
+	| YulLeave
+	| YulBreak
+	| YulContinue
+	| yulFunctionDefinition;
+
+yulBlock: YulLBrace yulStatement* YulRBrace;
+
+/**
+ * The declaration of one or more Yul variables with optional initial value.
+ * If multiple variables are declared, only a function call is a valid initial value.
+ */
+yulVariableDeclaration:
+    (YulLet variables+=YulIdentifier (YulAssign yulExpression)?)
+    | (YulLet variables+=YulIdentifier (YulComma variables+=YulIdentifier)* (YulAssign yulFunctionCall)?);
+
+/**
+ * Any expression can be assigned to a single Yul variable, whereas
+ * multi-assignments require a function call on the right-hand side.
+ */
+yulAssignment: yulPath YulAssign yulExpression | (yulPath (YulComma yulPath)+) YulAssign yulFunctionCall;
+
+yulIfStatement: YulIf cond=yulExpression body=yulBlock;
+
+yulForStatement: YulFor init=yulBlock cond=yulExpression post=yulBlock body=yulBlock;
+
+/**
+ * A Yul switch can only have one default case and must have at least one non-default case.
+ */
+yulSwitchCase: (YulDefault | (YulCase yulLiteral)) yulBlock;
+yulSwitchStatement: YulSwitch yulExpression | yulSwitchCase+;
```

Comment:
This doesn't enforce what the comment says

Reply:
And actually the comment was even wrong. Fixed.